Domain/Submission/src/main/java/tw/waterball/judgegirl/submission/domain/usecases/dto/SubmissionQueryParams.java

```diff
@@ -27,34 +27,22 @@
 public class SubmissionQueryParams {
     @Nullable
     private final Integer page;
-    private final int problemId;
-    private final String languageEnvName;
-    private final int studentId;
+    private final Map<String, String> queryParameters;
     private final Map<String, String> bagQueryParameters;
 
-    public SubmissionQueryParams(@Nullable Integer page, int problemId,
-                                 String languageEnvName, int studentId, Map<String, String> bagQueryParameters) {
+    public SubmissionQueryParams(@Nullable Integer page, Map<String, String> queryParameters,
+                                 Map<String, String> bagQueryParameters) {
         this.page = page;
-        this.problemId = problemId;
-        this.languageEnvName = languageEnvName;
-        this.studentId = studentId;
+        this.queryParameters = queryParameters;
```

Comment:
don't use Map, just make its type concrete

**we never use weak types when it's not needed**